Meta tags

altynbek07 opened the pull request.

Commit:
Созданы миграции создания таблиц для мета-тегов

Commit:
Созданы модели для мета-тегов

altynbek07 commented:
@naffiq Добавил модели мета-тегов

naffiq reviewed (approved):
Nice

naffiq commented on src/migrations/m180424_121053_create_meta_tag_translations_table.php:
```diff
@@ -18,8 +18,9 @@ public function up()
             'description' => $this->text()->comment('Описание'),
-            'keywords' => $this->text()->comment('Ключевые слова'),
-            'image' => $this->text()->comment('Изображение'),
+            'keywords' => $this->string()->comment('Ключевые слова'),
```
А почему опять к `string` переделал keywords?

altynbek07 replied:
@naffiq Я вот тут прочитал, что макс. длина тега `keywords` составляет 250 символов: http://www.bontonweb.com/seo/meta-tegi-descriptin-keywords-i-teg-title-i-ix-znachenie-dlya-optimizacii-sajta.html
По идее поле description тоже надо сделать типом `string`, так как рекомендуемая макс. длина 150 символов.
Ограничение по символам для полей `title`, `description`, `keywords` сделаю в следующем коммите в модели `MetaTagTranslation`

naffiq replied:
@altynbek07 Давай description пока не будем трогать, там и до 300 символов может быть (для мессенджеров и других шейров). Это по идее не только для сео

naffiq replied:
@altynbek07 Кстати, приведенная тобой статья устарела, не рекомендую под нее подстраиваться

altynbek07 replied:
@naffiq Мне вернуть `text` для поле `keywords`?

Commit:
Созданы поведение мета-тегов

naffiq commented on src/behaviors/MetaTagBehavior.php:
```diff
@@ -0,0 +1,109 @@
+     */
+    public function getMetaModel()
+    {
+        return MetaModel::findOne(['model' => $this->owner::className(), 'model_id' => $this->owner->id]);
```
Давай использовать `class` вместо `className` для обновления потом.

Commit:
Добавлен компонент мета-тегов
Исправлены ошибки в моделях мета-тегов

Commit:
Добавлен CRUD для MetaPage

Commit:
Добавлены переводы в моделях мета-тегов

altynbek07 commented:
@naffiq Done

Commit:
Создана виджет для подключение мета-тегов для моделей

Commit:
Добавлен поведение загрузки изображении

Commit:
Был обновлен поведение загрузки изображении:
ВАЖНО!
В поведениях, где используется класс '\mohorev\file\UploadBehavior', нужно заменить на '\naffiq\bridge\behaviors\BridgeUploadBehavior'.
В поведениях, где используется класс '\mohorev\file\UploadImageBehavior', нужно заменить на '\naffiq\bridge\behaviors\BridgeUploadImageBehavior'.

naffiq commented:
@altynbek07 

Давай зарезолвим конфликты и мердж уже)

Commit:
Добавлена изображение по-умолчанию для мета-тега og:image

naffiq committed:
Merge branch 'master' into meta-tags
